### PR TITLE
Do not record lineage for system datasets created in the default namespace.

### PR DESCRIPTION
**Pull request by sagarkapare**
JIRA: https://issues.cask.co/browse/CDAP-6924
JIRA: https://issues.cask.co/browse/CDAP-3732

Verified on cluster that queue dataset does not appear in the lineage.

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/writer/LineageWriterDatasetFramework.java [hunk 1]
```diff
@@ -151,7 +152,8 @@ public void deleteAllInstances(Id.Namespace namespaceId) throws DatasetManagemen
       // For non-system dataset, always perform authorization and lineage.
       AuthorizationEnforcer enforcer;
       DefaultDatasetRuntimeContext.DatasetAccessRecorder accessRecorder;
-      if (Id.Namespace.SYSTEM.equals(datasetInstanceId.getNamespace())) {
+      if (!DatasetsUtil.isUserDataset(datasetInstanceId)) {
+        LOG.info("SAGAR not writing lineage for {}", datasetInstanceId);
```
Comment (bdmogal) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/writer/LineageWriterDatasetFramework.java]: but can cdap still write? :)

Comment (sagarkapare) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/writer/LineageWriterDatasetFramework.java]: oops..

File: cdap-explore-client/src/main/java/co/cask/cdap/explore/client/ExploreFacade.java [hunk 2]
```diff
@@ -180,8 +180,14 @@ public void removeNamespace(Id.Namespace namespace) throws ExploreException, SQL
     handleExploreFuture(futureSuccess, "remove", "namespace", namespace.getId());
   }
 
+  //TODO: CDAP-4627 - Figure out a better way to identify system datasets in user namespaces
+  // Same check is done in DatasetsUtil.isUserDataset method.
   private boolean isDatasetExplorable(Id.DatasetInstance datasetInstance) {
```
Comment (bdmogal) [inline cdap-explore-client/src/main/java/co/cask/cdap/explore/client/ExploreFacade.java:185]: no way to share this code right?

**Comment by bdmogal** [comment]
qq about sharing the method, but I can't think of a place where we can put it. other than that, LGTM

**Comment by sagarkapare** [comment]
Release build is running here - http://builds.cask.co/browse/CDAP-RUT18-1

File: cdap-explore-client/src/main/java/co/cask/cdap/explore/client/ExploreFacade.java (hunk 2 above)
Comment (sagarkapare) [inline cdap-explore-client/src/main/java/co/cask/cdap/explore/client/ExploreFacade.java:185]: yeah, its a explore client which does not depend on the data-fabric.

**Comment by sagarkapare** [comment]
Build is green. Merging.

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetsUtil.java [hunk 3]
```diff
@@ -134,4 +134,13 @@ public static DatasetSpecification fixOriginalProperties(@Nullable DatasetSpecif
     }
     return spec.setOriginalProperties(props);
   }
+
+  //TODO: CDAP-4627 - Figure out a better way to identify system datasets in user namespaces
+  public static boolean isUserDataset(Id.DatasetInstance datasetInstanceId) {
```
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetsUtil.java:139]: This is really not a good way

Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetsUtil.java:139]: There's a JIRA to improve that right?
This PR is just to at least use this unclean way.

**Pull request merged**